cw_decred/lib/wallet.dart

```diff
@@ -0,0 +1,205 @@
+  Object get keys {
+    // throw UnimplementedError();
+    return {};
+  }
+
+  @override
+  late WalletAddressesBase walletAddresses;
+
+  // @override
+  // set isEnabledAutoGenerateSubaddress(bool value) {}
+
+  // @override
+  // bool get isEnabledAutoGenerateSubaddress => false;
+
+  @override
+  Future<void> connectToNode({required Node node}) async {
+    //throw UnimplementedError();
+  }
+
+  @action
+  @override
+  Future<void> startSync() async {
+    try {
+      this.spv.startSync();
+      syncStatus = this.spv.syncStatus();
+    } catch (e, stacktrace) {
+      print(stacktrace);
+      print(e.toString());
+      syncStatus = FailedSyncStatus();
+    }
+  }
+
+  @override
+  Future<PendingTransaction> createTransaction(Object credentials) async {
+    return this.spv.createTransaction(credentials);
+  }
+
+  int feeRate(TransactionPriority priority) {
+    try {
+      return this.spv.feeRate(priority.raw);
+    } catch(_) {
+      return 0;
+    }
+  }
+
+  @override
+  int calculateEstimatedFee(TransactionPriority priority, int? amount) {
+    if (priority is DecredTransactionPriority) {
+      return this.spv.calculateEstimatedFeeWithFeeRate(this.spv.feeRate(priority.raw), amount ?? 0);
+    }
+
+    return 0;
+  }
+
+  @override
+  Future<Map<String, TransactionInfo>> fetchTransactions() async {
+    // TODO: Use the transactionHistory.
+    final height = spv.height();
+    return this.spv.transactions(0, height);
+  }
+
+  @override
+  Future<void> save() async {
+    // What are we saving?
+  }
+
+  @override
+  Future<void> rescan({required int height}) async {
+    return spv.rescan(height);
+  }
+
+  @override
+  void close() {
+    this.spv.close();
+  }
+
+  @override
+  Future<void> changePassword(String password) async {
+    return this.spv.changePassword(password);
+  }
+
+  @override
+  String get password {
+    // throw UnimplementedError();
+    return "";
+  }
+
+  @override
+  Future<void>? updateBalance() async {
+    balance[CryptoCurrency.dcr] = this.spv.balance();
+  }
+
+  // @override
+  // void setExceptionHandler(void Function(FlutterErrorDetails) onError) => null;
+
+  Future<void> renameWalletFiles(String newWalletName) async {
+    final currentWalletPath = await pathForWallet(name: walletInfo.name, type: type);
+    final currentWalletFile = File(currentWalletPath);
+
+    final currentDirPath =
+        await pathForWalletDir(name: walletInfo.name, type: type);
+
+    // TODO: Stop the wallet, wait, and restart after.
+
+    // Copies current wallet files into new wallet name's dir and files
+    if (currentWalletFile.existsSync()) {
+      final newWalletPath = await pathForWallet(name: newWalletName, type: type);
+      await currentWalletFile.copy(newWalletPath);
+    }
+
+    // Delete old name's dir and files
+    await Directory(currentDirPath).delete(recursive: true);
+  }
+
+  // @override
+  // String signMessage(String message, {String? address = null}) => throw UnimplementedError();
```

Comment:
I see this is only used for "Robinhood", don't know what that is. dcrwallet supports message signing though; any reason we're not implementing?

Reply:
Enabling it.